dts/arm/st/h7/stm32h7.dtsi

```diff
@@ -862,6 +862,8 @@
 			interrupts = <127 0>;
 			status = "disabled";
 			#io-channel-cells = <1>;
+			temp-channel = <18>;
+			vref-channel = <19>;
```

Comment:
temp-channel and vref-channel are on different ADCs on different channels across the H7 family.
For H72x/73x, they are on ADC3 ch. 17 & 18.
For H74x/75x, they are on ADC3 ch. 18 & 19.
For H7Ax/7Bx, they are on ADC2 ch. 18 & 19.

If you take a look at stm32h723.dtsi, stm32h743.dtsi, stm32h745.dtsi, stm32h750.dtsi, or stm32h7a3.dtsi, they are all already defined.

But, there is no ADC3 on H7Ax/H7Bx, and ADC3 is already redefined for H72x. So I think we can keep these two lines in this file, but in that case please remove redundant definitions in stm32h743.dtsi, stm32h745.dtsi, stm32h750.dtsi.

I will take care to remove adc3 node in stm32h7a3.dtsi (I already have an open PR on ADC cleanup that I'll use).

Reply:
Appreciate the in-depth review!

After a closer look at the HAL and respective datasheets/RMs, I've also found that STM32H7A3 uses a different address for the `VREFINT` calibration data as well. Again, I'd falsely assumed all the STMH7 series shared the same parameters between them.

I've removed the above ADC channel properties from `stm32h7.dtsi`, as well as the `vref` node `io-channels` property, and added them to the individual SoC definitions, while also keeping their individual ADC node channel properties.

It feels a bit clearer to not have property defaults if these get overridden more than once for multiple diverging SoC specifications, even if it is a tad more verbose. IMO the best solution would be an intermediate common DTSI for the H72x/73x, H74x/75x, and H7Ax/7Bx sub-series, but that should probably be a separate PR.

Reply:
Good catch about the address for H7A, I hadn't checked that.